src/net.c

```diff
@@ -2900,6 +2900,19 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
         }
       }
 
+      if (resource == context->proxy_uri_resource &&
```

Comment:
As there may be other situations where sending an empty ACK is useful I suggest splitting this into two steps: First, set a flag that indicates sending the empty ACK if proxy_uri_resource and not reliable and then do the coap_send_ack/mid check/drop_it/mid update when this flag is set.
This flag might also be of use in the skip_handler code (but I am not sure about that).

Reply:
I have separately got something for mid de-duplication for responses as a pending PR.  I could not think of an easy, clean, way of doing mid de-duplication for non-proxy requests without a) always adding in an empty ack or b) keeping a copy of the response should a re-transmit re-request it.

As proxy requests will most of the time incur a delay in responding, adding in the empty ack early on was a good solution.

I'm happy with the concept of setting a flag that a is then tested for dropping an empty ack response from a handler which can then be used in future work, and will make the changes now.

Reply:
Code updated and pushed